source/simulator/melt.cc

```diff
@@ -1258,11 +1623,12 @@ namespace aspect
     variables.insert(variables.begin()+2,
                      VariableDeclaration<dim>(
                        "compaction pressure",
-                       (parameters.use_locally_conservative_discretization)
+                       (parameters.use_locally_conservative_discretization || use_discontinuous_p_c)
```

Comment:
Do you allow discontinuous normal pressure with melt? Please check. Even if so, remove the first part of this condition as you silently overwrite a setting from the input file.